app/controllers/application_controller.rb

```diff
@@ -1,13 +1,18 @@
 class ApplicationController < ActionController::Base
   include Pundit
   include Pagy::Backend
+  before_action :set_search
   before_action :configure_permitted_parameters, if: :devise_controller?
   around_action :switch_locale
 
   rescue_from Pundit::NotAuthorizedError do |exception|
     redirect_to root_url, alert: exception.message
   end
 
```

Comment:
all supplementary methods (`set_search`, `switch_locale`, `default_url_options`) should be in `private` or `protected` section